collectors/o365/index.js

```diff
@@ -15,7 +15,15 @@ const O365Collector = require('./o365_collector').O365Collector;
 exports.handler = O365Collector.makeHandler(function(event, context) {
     debug('input event: ', event);
     O365Collector.load().then(function(creds) {
-        var o365c = new O365Collector(context, creds);
-        o365c.handleEvent(event);
+        try {
+            var o365c = new O365Collector(context, creds);
+            o365c.handleEvent(event);
+        }
+        catch (err) {
```

Comment:
why do we need to catch and throw it again here?

Reply:
As per my understanding if we not add try catch then it will run next line too and then throw the error.
I will test and update.